include/repgp/repgp.h

```diff
@@ -59,155 +37,188 @@
 #include <stddef.h>
 #include <stdbool.h>
 #include <stdint.h>
-
 #include <json.h>
+
+#include <rnp/rnp_def.h>
 #include "repgp_def.h"
-#include "errors.h"
-
-struct rnp_key_store_t;
-typedef struct pgp_packet_t    pgp_packet_t;
-typedef struct pgp_cbdata_t    pgp_cbdata_t;
-typedef struct pgp_stream_t    pgp_stream_t;
-typedef struct pgp_reader_t    pgp_reader_t;
-typedef struct pgp_cryptinfo_t pgp_cryptinfo_t;
-typedef struct pgp_io_t        pgp_io_t;
-typedef struct pgp_key_t       pgp_key_t;
-typedef struct pgp_pubkey_t    pgp_pubkey_t;
-
-/** pgp_region_t */
-typedef struct pgp_region_t {
-    struct pgp_region_t *parent;
-    unsigned             length;
-    unsigned             readc; /* length read */
-    unsigned             last_read;
-    /* length of last read, only valid in deepest child */
-    unsigned indeterminate : 1;
-} pgp_region_t;
-
-void pgp_init_subregion(pgp_region_t *, pgp_region_t *);
-
-/** pgp_cb_ret_t */
-typedef enum { PGP_RELEASE_MEMORY, PGP_KEEP_MEMORY, PGP_FINISHED } pgp_cb_ret_t;
-typedef pgp_cb_ret_t pgp_cbfunc_t(const pgp_packet_t *, pgp_cbdata_t *);
+
+#define REPGP_HANDLE_NULL ((void *) 0)
+
+typedef struct pgp_packet_t pgp_packet_t;
+typedef struct pgp_stream_t pgp_stream_t;
+typedef struct repgp_handle repgp_handle_t;
+typedef struct repgp_io     repgp_io_t;
```

Comment:
These are inconsistent with the rest of the library, the tag and type name should be identical.

Reply:
I think I prefer keep it this way. I deliberately do not define type ``repgp_io_t``. I don't want to hide the fact that ``repgp_io`` is a struct

Reply:
Yeah, ok. That will be simpler. But I prefer to keep word struct in structure definitions. To make it clear what repgp_handler_t really is.

Reply:
I didn't really get what you meant here...

Reply:
I have some structure somewhere else in the code. In this structure I define variable as ``struct repgp_handler_t variable``.
What I mean is - it will stay that way (not ``repgp_handler_t variable;``)

Reply:
Ah I see what you meant. I don't think it's a big deal, but the current code base almost never does that (I see maybe 3 places) so it would be a bit inconsistent.

(Though you may want to ask ronald if librepgp should have a separate set of conventions anyways)